runtime/lua/vim/lsp/util.lua

```diff
@@ -85,9 +85,95 @@ local function get_border_size(opts)
   return { height = height, width = width }
 end
 
----@private
-local function split_lines(value)
-  return split(value, '\n', true)
+--- Splits the given text into lines according to LSP's definition of text
+--- lines. Quoting the specification: "To ensure that both client and server
+--- split the string into the same line representation the protocol specifies
+--- the following end-of-line sequences: '\n', '\r\n' and '\r'."
+---
+---@see |vim.lsp.util.split_lines_iter()|
+---@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocuments
+function M.split_lines(text, opts)
```

Comment:
Given we don't internally handle `\r` according to the specification, I'm not sure this is correct

Reply:
Are you talking about the function or the usages of it?

Reply:
I'm talking about we shouldn't be splitting lines on \r before sending to the server